server/service/integration_core_test.go

```diff
@@ -4533,14 +4534,63 @@ func (s *integrationTestSuite) TestUsers() {
 		Email:      ptr.String("extra@asd.com"),
 		Password:   ptr.String(userRawPwd),
 		GlobalRole: ptr.String(fleet.RoleObserver),
+		MFAEnabled: ptr.Bool(true),
 	}
+	// mailer isn't set up, which fails prior to silently ignoring MFA
+	s.DoJSON("POST", "/api/latest/fleet/users/admin", params, http.StatusBadRequest, &createResp)
+	params.MFAEnabled = nil
 	s.DoJSON("POST", "/api/latest/fleet/users/admin", params, http.StatusOK, &createResp)
 	assert.NotZero(t, createResp.User.ID)
 	assert.True(t, createResp.User.AdminForcedPasswordReset)
 	u := *createResp.User
 
-	// login as that user and check that teams info is empty
 	var loginResp loginResponse
+
+	// try MFA
+	mysql.ExecAdhocSQL(t, s.ds, func(db sqlx.ExtContext) error {
+		_, err := db.ExecContext(context.Background(), `UPDATE users SET mfa_enabled = TRUE WHERE id = ?`, u.ID)
+		return err
+	})
+	s.DoJSONWithoutAuth("POST", "/api/latest/fleet/sessions", sessionCreateRequest{Token: "foo"}, http.StatusUnauthorized, &loginResp)
+	// MFA unsupported client
+	s.DoJSONWithoutAuth("POST", "/api/latest/fleet/login", params, http.StatusBadRequest, &loginResp)
```

Comment:
params is `fleet. UserPayload`.

Reply:
Yep. This matches the existing usage on the same endpoint. I did a double-take as well as I implemented it but the payload satisfies the interface so it works fine.

Let me know if you feel strongly about it and I'll swap it out for the correct payload struct.